tests/test_permissions.py

```diff
@@ -191,7 +191,7 @@ def test_init(self) -> None:
 
     def test_init_invalid_perms(self) -> None:
         with pytest.raises(ValueError, match="'h' is not a valid permission name."):
-            PermissionOverwrite(h=True)
+            PermissionOverwrite(h=True)  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
Why is `# pyright: ignore` used here and not `# type: ignore`?

Reply:
we support python3.8 development environments, even if we don't need to, it makes the development process easier.

Reply:
https://github.com/DisnakeDev/disnake/pull/708/commits/5a2da47ad898ac49eed30d4df8f4adb9e8d9d5fa